modules/programs/zsh.nix

```diff
@@ -190,17 +202,22 @@ in
       envVarsStr = concatStringsSep "\n" (
         mapAttrsToList export config.home.sessionVariables
       );
+
+      zdotdir = "$HOME/" + cfg.dotDir;
     in mkIf cfg.enable {
       home.packages = with pkgs; [ zsh ]
         ++ optional cfg.enableCompletion nix-zsh-completions
         ++ optional cfg.oh-my-zsh.enable oh-my-zsh;
 
+      home.sessionVariables.ZDOTDIR = if cfg.dotDir != "" then zdotdir else null;
+
       home.file.".zshenv".text = ''
```

Comment:
What's the benefit of this though? The ~/.zshenv file needs to exist anyways

Reply:
Well, practically there is no difference, but since all zsh dot files are now located in another folder, it makes sense to me to have the main content of this file there as well and to have `~/.zshenv` just as an entry point. Motivation is the same as described [here](https://www.reddit.com/r/zsh/comments/3ubrdr/proper_way_to_set_zdotdir/cxe7ejm).